### DISCUSSION TITLE
What type of operating system structure if Unikraft was ported on RASP or other embedded system?

### DISCUSSION THREAD

**Discussion by freedom-ultimate** [unikraft/unikraft#922] [Q&A]
We are trying to porting Unikraft on bared metal device, not on VMMs.
So I think it is not Unikernel anymore.
It is libOS, or Monolithic kernel?
Could you tell the excat type of operating system structure if Unikraft was ported on RASP or other embedded system?

**Comment by nderjung** [comment]
Hi @freedom-ultimate. We apologise for the delay. We have [stopped using the GitHub Discussions feature](https://github.com/orgs/unikraft/discussions/806) in favour of our [community Discord server](https://bit.ly/UnikraftDiscord).

However, to answer your question: It will still be a unikernel. A unikernel is defined as a single-address space, single-process, monolithic kernel. It is an executable binary object capable of executing directly on a target hardware/platform combination, whether on virtualized hardware or physical hardware. Whilst monolithic in its final structure, it is composed via a library Operating System (libOS), making it minimal and bespoke and the runtime only performs the tasked defined by the user. We use the term monolithic still because this separates its composition from other OS types, such as the microkernel approach. A monolithic kernel is defined as one where the entire OS is working in kernel space.
